### ISSUE TITLE
Undefined symbol error when importing module into Lua 5.1

### ISSUE THREAD

**Issue by MudcrabDev** [khvzak/mlua#25]
Lua 5.1 is unable to import a module built with mlua even though cargo builds it without error.
```
Lua 5.1.5  Copyright (C) 1994-2012 Lua.org, PUC-Rio
> require("librust_module")
error loading module 'librust_module' from file './librust_module.so':
	./librust_module.so: undefined symbol: luaopen_librust_module
stack traceback:
	[C]: ?
	[C]: in function 'require'
	stdin:1: in main chunk
	[C]: ?
```
My Cargo.toml (taken and modified from the example in the repo):
```
[package]
name = "rust_module"
version = "0.0.0"
authors = ["Aleksandr Orlenko <zxteam@pm.me>"]
edition = "2018"

[lib]
crate-type = ["cdylib"]

[dependencies]
lua-src = "542.0.0"
mlua = { version = "0.4.2", features = ["lua51", "vendored", "module"] }
mlua_derive = "0.4.0"
```
My lib.rs:
```
use mlua::prelude::*;

fn sum(_: &Lua, (a, b): (i64, i64)) -> LuaResult<i64> {
    Ok(a + b)
}

fn used_memory(lua: &Lua, _: ()) -> LuaResult<usize> {
    Ok(lua.used_memory())
}

#[mlua_derive::lua_module]
fn rust_module(lua: &Lua) -> LuaResult<LuaTable> {
    let exports = lua.create_table()?;
    exports.set("sum", lua.create_function(sum)?)?;
    exports.set("used_memory", lua.create_function(used_memory)?)?;
    Ok(exports)
}
```
I'm building on Pop! OS Linux 20.04. Rust version: `rustc 1.49.0 (e1884a8e3 2020-12-29)`
Built with: `cargo build --release`
I am a novice programmer so if you could please ELI5 I would greatly appreciate it.

**Comment by khvzak** [comment]
Seems this is your post in reddit as well? :)  https://www.reddit.com/r/rust/comments/kl910i/lua_cant_load_mlua_built_module/ 

The main issue there is a module name `librust_module`.  Could you try to `require("rust_module")`? The module name must match with the function name (`rust_module` is this case).
And therefore shared library file should be renamed (or symlinked) as well.

**Comment by MudcrabDev** [comment]
Renaming it to rust_module.so did the trick! Thank you for the help. :)
And yes that was my reddit post. I will update it with your answer for anyone else who runs into the same problem.
